### ISSUE TITLE
[Bug]: conky v1.16.0 doesn't run in KDE/X11 even with default config

### ISSUE THREAD

**Issue by iDarkTemplar** [brndnmtthws/conky#1338]
### What happened?

After building conky 1.16.0 it couldn't start with my config. After that I tried default config from data/conky.conf from conky repo, it still didn't work. Only error output is following:

```
$ conky -c ./data/conky.conf 
conky: drawing to single buffer
conky: invalid setting of type 'table'
```

Environment: KDE/Plasma5 on X11, Gentoo amd64.

I've tried bisecting, but couldn't find exact commit.
<details>
  <summary>Cmake options while bisecting</summary>

```
cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_ARGB=yes -DBUILD_X11=yes -DBUILD_XDAMAGE=yes -DBUILD_XDBE=yes -DBUILD_XSHAPE=yes -DOWN_WINDOW=yes -DBUILD_APCUPSD=no -DBUILD_AUDACIOUS=no -DBUILD_BUILTIN_CONFIG=yes -DBUILD_CMUS=no -DBUILD_CURL=no -DBUILD_DOCS=no -DBUILD_EXTRAS=yes -DBUILD_HDDTEMP=yes -DBUILD_HTTP=no -DBUILD_I18N=yes -DBUILD_IBM=yes -DBUILD_ICAL=no -DBUILD_ICONV=yes -DBUILD_IMLIB2=yes -DBUILD_INTEL_BACKLIGHT=no -DBUILD_IOSTATS=yes -DBUILD_IPV6=yes -DBUILD_IRC=no -DBUILD_JOURNAL=no -DBUILD_LUA_CAIRO=no -DBUILD_LUA_IMLIB2=yes -DBUILD_LUA_RSVG=no -DBUILD_MATH=yes -DBUILD_MOC=no -DBUILD_MPD=no -DBUILD_MYSQL=no -DBUILD_NCURSES=yes -DBUILD_NVIDIA=yes -DBUILD_OLD_CONFIG=yes -DBUILD_PORT_MONITORS=yes -DBUILD_PULSEAUDIO=yes -DBUILD_RSS=no -DBUILD_WLAN=yes -DBUILD_XFT=yes -DBUILD_XINERAMA=yes -DBUILD_XMMS2=no -DDOC_PATH=/usr/share/doc/conky-1.16.0 -DMAINTAINER_MODE=no -DRELEASE=yes -DPython3_EXECUTABLE=/usr/bin/python3.10 -DCMAKE_BUILD_TYPE=Debug -DBUILD_WAYLAND=yes
```
</details>

<details>
  <summary>Bisect results</summary>

```
$ git bisect skip
There are only 'skip'ped commits left to test.
The first bad commit could be any of:
343f4545cd50cf7ae64739a5a6b68148bcbd6f65
0b00ff487f7ad2a0abbc3b9315df369db1c99a54
e5413a22d7a5393c70bcc398cc2f372866cd2a51
03d9c25491f780f86e235a3ae43f5efdf76ec5c2
20ac59b0b161122571844bef2dd5906c5cc2606a
65f4d4f7ebe88a4f75e049b21496828f996110bc
d856e397d728ea3cb7fe1c59a1e8b17403955963
d183ad3f6fbc4e1b6d2e242409237fc227748407
e914128c323fe124092b3d70b9936ea1aa5cae5a
35b874de6535fd91baec315364d77b241924fb4b
5859d7674189c10054f4e18593f25d1c7d021e88
2a4f5216c17c58ff68318c6987ce7fce5165abe5
4c71e4527b637deae04dbc919dd3ea6d2104d7f5
8973b582995c649cf87b038ec60817faa5d86dfb
9ce6ffbb87cbe68af62e356369dfd776b7c4a0e5
a30cc64a9848e15078cbd60dedc841c7a41cb620
d11742ae184a9d70415bbf8ede917b13593e2fa5
1eb88aeb433c2dfcad8cd44b9b2bb03a646f2c91
We cannot bisect more!
```
</details>

<details>
  <summary>Bisect full log</summary>

```
$ git bisect log
# bad: [5f0ed0d72d643bcf8711ec3fe23cdfaf1ecd6855] Bump version.
# good: [70d7d5c796e3e84e674705ffc6aaf7a0d6c5b109] Bump version for next release.
git bisect start 'v1.16.0' 'v1.15.0'
# good: [93c38c60af7791444f585fd4e91b6e3075b02c6e] docs/variables.yaml: A better description for memlaundry.
git bisect good 93c38c60af7791444f585fd4e91b6e3075b02c6e
# skip: [2a4f5216c17c58ff68318c6987ce7fce5165abe5] x11, gui: update display-x11.cc to account for some code moved into gui.h/gui.cc
git bisect skip 2a4f5216c17c58ff68318c6987ce7fce5165abe5
# bad: [09da5bbf0c99866822f2d91a7a710642a382a381] Bump web deps.
git bisect bad 09da5bbf0c99866822f2d91a7a710642a382a381
# skip: [5859d7674189c10054f4e18593f25d1c7d021e88] core: fix misbehavior in specials handling that fails badly with Wayland backend
git bisect skip 5859d7674189c10054f4e18593f25d1c7d021e88
# skip: [e5413a22d7a5393c70bcc398cc2f372866cd2a51] config: add out_to_wayland to bool setting list for config converter
git bisect skip e5413a22d7a5393c70bcc398cc2f372866cd2a51
# bad: [cf16110c7a9f7362cd4f0bbe8c348cffd06a206b] Merge branch 'bi4k8-wayland-gui-backend'
git bisect bad cf16110c7a9f7362cd4f0bbe8c348cffd06a206b
# skip: [d183ad3f6fbc4e1b6d2e242409237fc227748407] core: fix segfault when no display outputs are enabled
git bisect skip d183ad3f6fbc4e1b6d2e242409237fc227748407
# skip: [9ce6ffbb87cbe68af62e356369dfd776b7c4a0e5] wl: declare desktop/monitor info functions as weak symbols
git bisect skip 9ce6ffbb87cbe68af62e356369dfd776b7c4a0e5
# skip: [8973b582995c649cf87b038ec60817faa5d86dfb] gui: remove background_color from gui.cc and window_type/window_hints from gui.h
git bisect skip 8973b582995c649cf87b038ec60817faa5d86dfb
# bad: [a6c52ee73f96ee733eae74ed3a260046b18496e4] wayland: clamp text width
git bisect bad a6c52ee73f96ee733eae74ed3a260046b18496e4
# skip: [20ac59b0b161122571844bef2dd5906c5cc2606a] doc: document out_to_wayland setting
git bisect skip 20ac59b0b161122571844bef2dd5906c5cc2606a
# skip: [d856e397d728ea3cb7fe1c59a1e8b17403955963] core: fix right-alignment with wayland
git bisect skip d856e397d728ea3cb7fe1c59a1e8b17403955963
# skip: [4c71e4527b637deae04dbc919dd3ea6d2104d7f5] wayland: implement buffer scaling
git bisect skip 4c71e4527b637deae04dbc919dd3ea6d2104d7f5
# skip: [343f4545cd50cf7ae64739a5a6b68148bcbd6f65] wayland: rework window destruction/resizing
git bisect skip 343f4545cd50cf7ae64739a5a6b68148bcbd6f65
# bad: [1eb88aeb433c2dfcad8cd44b9b2bb03a646f2c91] x11, imlib2: early-exit in some X functions if X is not available at runtime
git bisect bad 1eb88aeb433c2dfcad8cd44b9b2bb03a646f2c91
# skip: [0b00ff487f7ad2a0abbc3b9315df369db1c99a54] wayland: respond to xdg_wm_base.ping
git bisect skip 0b00ff487f7ad2a0abbc3b9315df369db1c99a54
# skip: [03d9c25491f780f86e235a3ae43f5efdf76ec5c2] wayland: fix crash after 5 minutes
git bisect skip 03d9c25491f780f86e235a3ae43f5efdf76ec5c2
# skip: [d11742ae184a9d70415bbf8ede917b13593e2fa5] build, colours, core, gui, wayland, wl, x11: implement wayland display backend
git bisect skip d11742ae184a9d70415bbf8ede917b13593e2fa5
# skip: [65f4d4f7ebe88a4f75e049b21496828f996110bc] colours: define manually_get_x11_color for usage outside X11
git bisect skip 65f4d4f7ebe88a4f75e049b21496828f996110bc
# skip: [35b874de6535fd91baec315364d77b241924fb4b] wayland: switch to zwlr_layer_shell_v1 and implement proper window placement
git bisect skip 35b874de6535fd91baec315364d77b241924fb4b
# skip: [e914128c323fe124092b3d70b9936ea1aa5cae5a] colours: parse X11 colors under Wayland
git bisect skip e914128c323fe124092b3d70b9936ea1aa5cae5a
# good: [4622c0e336f96841d7235a61d169e93fbef220df] Change assert in set_font to check w/ debug.
git bisect good 4622c0e336f96841d7235a61d169e93fbef220df
# skip: [a30cc64a9848e15078cbd60dedc841c7a41cb620] x11: update x11.cc for functionality moved into gui.cc
git bisect skip a30cc64a9848e15078cbd60dedc841c7a41cb620
# only skipped commits left to test
# possible first bad commit: [1eb88aeb433c2dfcad8cd44b9b2bb03a646f2c91] x11, imlib2: early-exit in some X functions if X is not available at runtime
# possible first bad commit: [a30cc64a9848e15078cbd60dedc841c7a41cb620] x11: update x11.cc for functionality moved into gui.cc
# possible first bad commit: [8973b582995c649cf87b038ec60817faa5d86dfb] gui: remove background_color from gui.cc and window_type/window_hints from gui.h
# possible first bad commit: [2a4f5216c17c58ff68318c6987ce7fce5165abe5] x11, gui: update display-x11.cc to account for some code moved into gui.h/gui.cc
# possible first bad commit: [5859d7674189c10054f4e18593f25d1c7d021e88] core: fix misbehavior in specials handling that fails badly with Wayland backend
# possible first bad commit: [d183ad3f6fbc4e1b6d2e242409237fc227748407] core: fix segfault when no display outputs are enabled
# possible first bad commit: [d856e397d728ea3cb7fe1c59a1e8b17403955963] core: fix right-alignment with wayland
# possible first bad commit: [20ac59b0b161122571844bef2dd5906c5cc2606a] doc: document out_to_wayland setting
# possible first bad commit: [e5413a22d7a5393c70bcc398cc2f372866cd2a51] config: add out_to_wayland to bool setting list for config converter
# possible first bad commit: [343f4545cd50cf7ae64739a5a6b68148bcbd6f65] wayland: rework window destruction/resizing
# possible first bad commit: [0b00ff487f7ad2a0abbc3b9315df369db1c99a54] wayland: respond to xdg_wm_base.ping
# possible first bad commit: [03d9c25491f780f86e235a3ae43f5efdf76ec5c2] wayland: fix crash after 5 minutes
# possible first bad commit: [65f4d4f7ebe88a4f75e049b21496828f996110bc] colours: define manually_get_x11_color for usage outside X11
# possible first bad commit: [e914128c323fe124092b3d70b9936ea1aa5cae5a] colours: parse X11 colors under Wayland
# possible first bad commit: [35b874de6535fd91baec315364d77b241924fb4b] wayland: switch to zwlr_layer_shell_v1 and implement proper window placement
# possible first bad commit: [4c71e4527b637deae04dbc919dd3ea6d2104d7f5] wayland: implement buffer scaling
# possible first bad commit: [9ce6ffbb87cbe68af62e356369dfd776b7c4a0e5] wl: declare desktop/monitor info functions as weak symbols
# possible first bad commit: [d11742ae184a9d70415bbf8ede917b13593e2fa5] build, colours, core, gui, wayland, wl, x11: implement wayland display backend
# skip: [a30cc64a9848e15078cbd60dedc841c7a41cb620] x11: update x11.cc for functionality moved into gui.cc
git bisect skip a30cc64a9848e15078cbd60dedc841c7a41cb620
# only skipped commits left to test
# possible first bad commit: [1eb88aeb433c2dfcad8cd44b9b2bb03a646f2c91] x11, imlib2: early-exit in some X functions if X is not available at runtime
# possible first bad commit: [a30cc64a9848e15078cbd60dedc841c7a41cb620] x11: update x11.cc for functionality moved into gui.cc
# possible first bad commit: [8973b582995c649cf87b038ec60817faa5d86dfb] gui: remove background_color from gui.cc and window_type/window_hints from gui.h
# possible first bad commit: [2a4f5216c17c58ff68318c6987ce7fce5165abe5] x11, gui: update display-x11.cc to account for some code moved into gui.h/gui.cc
# possible first bad commit: [5859d7674189c10054f4e18593f25d1c7d021e88] core: fix misbehavior in specials handling that fails badly with Wayland backend
# possible first bad commit: [d183ad3f6fbc4e1b6d2e242409237fc227748407] core: fix segfault when no display outputs are enabled
# possible first bad commit: [d856e397d728ea3cb7fe1c59a1e8b17403955963] core: fix right-alignment with wayland
# possible first bad commit: [20ac59b0b161122571844bef2dd5906c5cc2606a] doc: document out_to_wayland setting
# possible first bad commit: [e5413a22d7a5393c70bcc398cc2f372866cd2a51] config: add out_to_wayland to bool setting list for config converter
# possible first bad commit: [343f4545cd50cf7ae64739a5a6b68148bcbd6f65] wayland: rework window destruction/resizing
# possible first bad commit: [0b00ff487f7ad2a0abbc3b9315df369db1c99a54] wayland: respond to xdg_wm_base.ping
# possible first bad commit: [03d9c25491f780f86e235a3ae43f5efdf76ec5c2] wayland: fix crash after 5 minutes
# possible first bad commit: [65f4d4f7ebe88a4f75e049b21496828f996110bc] colours: define manually_get_x11_color for usage outside X11
# possible first bad commit: [e914128c323fe124092b3d70b9936ea1aa5cae5a] colours: parse X11 colors under Wayland
# possible first bad commit: [35b874de6535fd91baec315364d77b241924fb4b] wayland: switch to zwlr_layer_shell_v1 and implement proper window placement
# possible first bad commit: [4c71e4527b637deae04dbc919dd3ea6d2104d7f5] wayland: implement buffer scaling
# possible first bad commit: [9ce6ffbb87cbe68af62e356369dfd776b7c4a0e5] wl: declare desktop/monitor info functions as weak symbols
# possible first bad commit: [d11742ae184a9d70415bbf8ede917b13593e2fa5] build, colours, core, gui, wayland, wl, x11: implement wayland display backend
# skip: [a30cc64a9848e15078cbd60dedc841c7a41cb620] x11: update x11.cc for functionality moved into gui.cc
git bisect skip a30cc64a9848e15078cbd60dedc841c7a41cb620
# only skipped commits left to test
# possible first bad commit: [1eb88aeb433c2dfcad8cd44b9b2bb03a646f2c91] x11, imlib2: early-exit in some X functions if X is not available at runtime
# possible first bad commit: [a30cc64a9848e15078cbd60dedc841c7a41cb620] x11: update x11.cc for functionality moved into gui.cc
# possible first bad commit: [8973b582995c649cf87b038ec60817faa5d86dfb] gui: remove background_color from gui.cc and window_type/window_hints from gui.h
# possible first bad commit: [2a4f5216c17c58ff68318c6987ce7fce5165abe5] x11, gui: update display-x11.cc to account for some code moved into gui.h/gui.cc
# possible first bad commit: [5859d7674189c10054f4e18593f25d1c7d021e88] core: fix misbehavior in specials handling that fails badly with Wayland backend
# possible first bad commit: [d183ad3f6fbc4e1b6d2e242409237fc227748407] core: fix segfault when no display outputs are enabled
# possible first bad commit: [d856e397d728ea3cb7fe1c59a1e8b17403955963] core: fix right-alignment with wayland
# possible first bad commit: [20ac59b0b161122571844bef2dd5906c5cc2606a] doc: document out_to_wayland setting
# possible first bad commit: [e5413a22d7a5393c70bcc398cc2f372866cd2a51] config: add out_to_wayland to bool setting list for config converter
# possible first bad commit: [343f4545cd50cf7ae64739a5a6b68148bcbd6f65] wayland: rework window destruction/resizing
# possible first bad commit: [0b00ff487f7ad2a0abbc3b9315df369db1c99a54] wayland: respond to xdg_wm_base.ping
# possible first bad commit: [03d9c25491f780f86e235a3ae43f5efdf76ec5c2] wayland: fix crash after 5 minutes
# possible first bad commit: [65f4d4f7ebe88a4f75e049b21496828f996110bc] colours: define manually_get_x11_color for usage outside X11
# possible first bad commit: [e914128c323fe124092b3d70b9936ea1aa5cae5a] colours: parse X11 colors under Wayland
# possible first bad commit: [35b874de6535fd91baec315364d77b241924fb4b] wayland: switch to zwlr_layer_shell_v1 and implement proper window placement
# possible first bad commit: [4c71e4527b637deae04dbc919dd3ea6d2104d7f5] wayland: implement buffer scaling
# possible first bad commit: [9ce6ffbb87cbe68af62e356369dfd776b7c4a0e5] wl: declare desktop/monitor info functions as weak symbols
# possible first bad commit: [d11742ae184a9d70415bbf8ede917b13593e2fa5] build, colours, core, gui, wayland, wl, x11: implement wayland display backend
```
</details>

### Version

1.16.0

### Which OS/distro are you seeing the problem on?

Gentoo

**Comment by iDarkTemplar** [comment]
Forgot to mention, 1.15.0 works good on same setup.

**Comment by brndnmtthws** [comment]
Uh oh, it seems I've botched this. Let me try and get it sorted out ASAP.

**Comment by brndnmtthws** [comment]
@bi4k8 can you please take a look?

**Comment by brndnmtthws** [comment]
This should be (at least partially) addressed by #1341.

**Comment by iDarkTemplar** [comment]
Linked pull request fixes this issue for me.

**Comment by iDarkTemplar** [comment]
Closing as fixed in 1.16.1.
